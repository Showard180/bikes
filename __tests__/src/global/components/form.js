import 'react-native';
import React from 'react';
import Form from '../../../../src/global/components/form';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders', () => {
  const props = {
    shards: []
  }

  const tree = renderer.create(
    <Form {...props} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders an input when one given in props', () => {
  const props = {
    shards: [
      {
        type: 'text',
        'function': 'onChange',
        text: 'This is a test',
        title: 'Write here for a test'
      }
    ],
    functions: {
      onChange: () => null
    }
  }

  const tree = renderer.create(
    <Form {...props} />
  );
  expect(tree).toMatchSnapshot();
});

test('renders an error when given one in props', () => {
  const props = {
    shards: [],
    err: 'This is a test error'
  };

  const tree = renderer.create(
    <Form {...props} />
  );
  expect(tree).toMatchSnapshot();
})
