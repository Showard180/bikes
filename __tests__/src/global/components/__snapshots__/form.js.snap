// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `<View />`;

exports[`renders an error when given one in props 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "alignSelf": "center",
        "color": "red",
        "marginBottom": 30,
      }
    }
  >
    This is a test error
  </Text>
</View>
`;

exports[`renders an input when one given in props 1`] = `
<View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "marginLeft": 15,
        }
      }
    >
      Write here for a test
    </Text>
    <View
      style={
        Object {
          "backgroundColor": "transparent",
          "height": 40,
          "marginVertical": 10,
        }
      }
    >
      <TextInput
        keyboardType={undefined}
        onChangeText={[Function]}
        placeholder="This is a test"
        secureTextEntry={undefined}
        style={
          Object {
            "backgroundColor": "white",
            "borderWidth": 1,
            "flex": 1,
            "paddingHorizontal": 10,
          }
        }
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
</View>
`;
